### ISSUE TITLE
ActionController::InvalidAuthenticityToken when submitting a form via button with `formmethod: :delete`

### ISSUE THREAD

**Issue by pokonski** [rails/rails#52290]
### Steps to reproduce
1. Create a new app with `rails _7.2.0.beta2_  new csrf_repro -d sqlite3`
2. Add scaffold with `rails g scaffold Products`
3. Replace `app/views/products/show.html.erb` with following:
  ```erb
  <%= form_with model: @product do |f| %>
    <%= f.button "Remove product", formmethod: :delete %>
  <% end %>
  ```

4. Disable JavaScript
5. Navigate to http://localhost:3000/products/new
6. Create new product
7. Click "Remove product"

Repo with the project repro: https://github.com/pokonski/csrf_repro

### Expected behavior
products#destroy action should be properly executed and product removed. 

The docs suggest this is possible (https://guides.rubyonrails.org/form_helpers.html#how-do-forms-with-patch-put-or-delete-methods-work-questionmark) b

### Actual behavior
`ActionController::InvalidAuthenticityToken` is raised instead when Hotwire is disabled (via Javascript disable or just removing it from application.js)

The difference seems to be in how `_method` param is submitted:

- with JS and Hotwire enabled there's only a single param:

<img width="844" alt="image" src="https://github.com/rails/rails/assets/135402/dc016a29-1b5d-411e-a6ed-31c84acbb4e0">

- without JS/Hotwire the `_method` param appears twice with both the original method (`patch`) and `delete`

<img width="695" alt="image" src="https://github.com/rails/rails/assets/135402/9ca2144a-037c-41c8-85ac-7e7bea7b856a">

Issue is present in both Firefox 127 and Chrome 126

### System configuration
**Rails version**: 7.2.0-beta2

**Ruby version**: ruby 3.2.0 (2022-12-25 revision a528908271) [arm64-darwin23]

**Comment by pokonski** [comment]
After debugging `valid_authenticity_token?` method in `lib/action_controller/metal/request_forgery_protection.rb` it seems the token check fails on `valid_per_form_csrf_token?(csrf_token)` for the DELETE request, but succeeds when the method is `PATCH`

**Comment by pokonski** [comment]
Okay, so disabling per-form tokens fixes the issue: `config.action_controller.per_form_csrf_tokens = false`

Should that be documented in the guides about button's `formmethod` attribute? If so I can prepare a fix to the docs

**Comment by Jay0921** [comment]
This seems like a bug to me. Not only is the token generated using the wrong identifier (it uses the method in the form rather than `formmethod`), but the request is also verified against a global CSRF token instead of a form-specific CSRF token.

I looked into the code, and fixing this might not be straightforward. The `form_with` method doesn’t have access to the `formmethod` attribute within the block. Moving the `formmethod` to the `form_with` method could be a simpler solution, but I’m not sure about the side effects. Let’s wait for the core team’s response.


If we can access `formmethod` here, we can override the `html_options`.
https://github.com/rails/rails/blob/ef087627f53fac581e6120332c61e96d5e68628c/actionview/lib/action_view/helpers/form_helper.rb#L774-L779

**Comment by pokonski** [comment]
I think this could be approached in the opposite direction. When calling `button` with a `:formmethod` attribute, it could switch the form's authentication_token to a global one instead of form. Because we can't have multiple tokens targetting different methods in one form

**Comment by Jay0921** [comment]
@pokonski You are right. I wasn't aware of the use cases for multiple submit buttons. In that case, moving `formmethod` to the `form_with` method won't work.

**Comment by matchu** [comment]
I ran into this as well! Still happening in Rails 8.0.2.

**Comment by rnestler** [comment]
I could fix it locally by adding an additional
```ruby
hidden_field_tag :authenticity_token, form_authenticity_token
```

in the form.

**Comment by zzak** [comment]
I looked into this, I don't think there is anything we can do so I've added PR with docs.

**Comment by schmijos** [comment]
This issue is not only with [`formmethod`](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/input/submit#formmethod) but also with [`formaction`](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/input/submit#formaction). Both are standard attributes on images, buttons and submit buttons.

**Comment by zzak** [comment]
@schmijos Do you want to take over the docs PR and improve on what I've added?
